src/tests/ftest/util/dmg_utils.py

```diff
@@ -702,6 +823,129 @@ def get_pool_uuid_service_replicas_from_stdout(stdout_str):
     return uuid, svc
 
 
+def query_smd_info(stdout_str):
```

Comment:
As we discussed, a more generic method for parsing the CmdResult objects probably makes sense and would reduce the amount of code needed for each helper-method-info-gathering method.